lib/layout/forceDirectedLayout.js

```diff
@@ -144,111 +158,92 @@ var springForce = function(edge) {
-      }
-      if(!ports[j].get('isPinned')) {
-        ports[j].set('position', new CardinalPortPosition({direction: cardinalDirection, percentage: posPercentage}));
-      }
-    }
-  }
-};
-
-var layoutPorts = function(graph) {
-  var verts = graph.get('vertices');
-  var maxPortsPerSide = 6;
-  for(var i = 0; i < verts.length; i++) {
-    var ins = verts[i].get('inputs');
-    var outs = verts[i].get('outputs');
-    portLayoutHelper(ins, CardinalDirection.WEST, maxPortsPerSide);
-    portLayoutHelper(outs, CardinalDirection.EAST, maxPortsPerSide);
+  //var minLimit = 0.1;
+  var toIndex;
+  var fromIndex;
+  if(vertexTo.get('isPinned') === false && vertexFrom.get('isPinned') === false){
+    toIndex = vertices.indexOf(vertexTo);
+    vectors[toIndex].x += repulsionX;
+    vectors[toIndex].y += repulsionY;
+    vectors[toIndex].count += 1;
+    fromIndex = vertices.indexOf(vertexFrom);
+    vectors[fromIndex].x -= repulsionX;
+    vectors[fromIndex].y -= repulsionY;
+    vectors[fromIndex].count += 1;
+
+
+  }else if(vertexTo.get('isPinned') === false){
+
+    toIndex = vertices.indexOf(vertexTo);
+    vectors[toIndex].x += (repulsionX * 2);
+    vectors[toIndex].y += (repulsionY * 2);
+    vectors[toIndex].count ++;
+
+  }else if(vertexFrom.get('isPinned') === false){
+    fromIndex = vertices.indexOf(vertexFrom);
+    vectors[fromIndex].x -= (repulsionX * 2);
+    vectors[fromIndex].y -= (repulsionY * 2);
+    vectors[fromIndex].count ++;
   }
 };
 
-//Based partially on https://code.google.com/p/jung/source/browse/branches/
-//guava/jung/jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/
-//layout/SpringLayout.java?r=19
-
-/**
- * Defines force directed layout algorithm
- *
- * @class forceDirectedLayout
- *
- */
-
-/**
- * Takes the graph and re aligns the nodes
- *
- * Returns nothing
- *
- * @method layOutGraph
- * @param  {Graph} graph
- * @return nothing
- */
-
-var layOutGraph = function(graph) {
+ /**
+     * Takes the graph and re aligns the nodes
+     *
+     * Returns nothing
+     *
+     * @method layOutGraph
+     * @param  {Graph} graph
+     * @return nothing
+     */
+
+var layOutGraph = function(graph){
+  dummyLayout(graph);
   mGraph = graph;
+  vertices = graph.get('vertices');
+  vectors = [];
   var edgeList = graph.get('edges');
   var runner = true;
-  layoutPorts(mGraph);
-  while (runner) {
+
+  while(runner){
+    for (var a = 0; a < vertices.length; a++){
+      vectors.push({x: 0, y: 0, count: 0});
+    }
     change = false;
-    for (var i = 0; i < edgeList.length; i++) {
+    for(var i = 0; i < edgeList.length; i++){
       springForce(edgeList[i]);
     }
-    if (change === false) {
+    for (a = 0; a < vertices.length; a++){
+      if(vectors[a].x >= minLimit || vectors[a].y >= minLimit || vectors[a].x <= -minLimit || vectors[a].y <= -minLimit){
+        change = true;
+      }
+      if(vectors[a].x > 1000){
+        vertices[a].get('position').x = vertices[a].get('position').x + 1000;
```

Comment:
repeating getting of attribute. very inefficient, and trusting that this external function will always be fast.